dbms/src/Interpreters/ThreadStatusExt.cpp

```diff
@@ -41,6 +42,8 @@ void ThreadStatus::attachQueryContext(Context & query_context_)
         if (!thread_group->global_context)
             thread_group->global_context = global_context;
     }
+
+    initQueryProfiler();
```

Comment:
It will probably set up query profiler only in first query execution thread.
Look at my recent changes related to `setpriority` in this file.